server/apps/publish/content/remove_tests.py

```diff
@@ -0,0 +1,635 @@
+# -*- coding: utf-8; -*-
+#
+# This file is part of Superdesk.
+#
+# Copyright 2013, 2014 Sourcefabric z.u. and contributors.
+#
+# For the full copyright and license information, please see the
+# AUTHORS and LICENSE files distributed with this source code, or
+# at https://www.sourcefabric.org/superdesk/license
+
+from datetime import timedelta
+import os
+import json
+
+from eve.utils import config
+from eve.versioning import versioned_id_field
+
+from apps.archive.common import insert_into_versions, is_takes_package, ITEM_OPERATION
+from apps.packages.package_service import PackageService
+from apps.packages.takes_package_service import TakesPackageService
+from .common import WIRE
+from superdesk.publish.subscribers import SUBSCRIBER_TYPES
+from apps.validators import ValidatorsPopulateCommand
+from settings import URL_PREFIX
+from superdesk.metadata.packages import RESIDREF, GROUPS
+from test_factory import SuperdeskTestCase
+from apps.publish import init_app, RemoveExpiredPublishContent
+from apps.legal_archive import LEGAL_ARCHIVE_NAME, LEGAL_ARCHIVE_VERSIONS_NAME, LEGAL_PUBLISH_QUEUE_NAME
+from superdesk.utc import utcnow
+from superdesk import get_resource_service
+from apps.archive.archive import SOURCE as ARCHIVE
+from superdesk.metadata.item import ITEM_STATE, CONTENT_STATE, ITEM_TYPE, CONTENT_TYPE
+
+ARCHIVE_PUBLISH = 'archive_publish'
+ARCHIVE_CORRECT = 'archive_correct'
+ARCHIVE_KILL = 'archive_kill'
+
+PUBLISH_QUEUE = 'publish_queue'
+PUBLISHED = 'published'
+
+
+class RemoveExpiredFromPublishedCollection(SuperdeskTestCase):
+    def setUp(self):
+        super().setUp()
+        self._init_data()
+
+        self.app.data.insert('vocabularies', self.vocabularies)
+        self.app.data.insert('subscribers', self.subscribers)
+        self.app.data.insert(ARCHIVE, self.articles)
+
+        self.filename = os.path.join(os.path.abspath(os.path.dirname(__file__)), "validators.json")
+        self.json_data = [
+            {"_id": "kill_text", "act": "kill", "type": "text", "schema": {"headline": {"type": "string"}}},
+            {"_id": "publish_text", "act": "publish", "type": "text", "schema": {}},
+            {"_id": "correct_text", "act": "correct", "type": "text", "schema": {}},
+            {"_id": "publish_composite", "act": "publish", "type": "composite", "schema": {}},
+        ]
+
+        with open(self.filename, "w+") as file:
+            json.dump(self.json_data, file)
+        init_app(self.app)
+        ValidatorsPopulateCommand().run(self.filename)
+
+        self.package_service = PackageService()
+
+    def tearDown(self):
+        super().tearDown()
+        if self.filename and os.path.exists(self.filename):
+            os.remove(self.filename)
+
+    def test_can_remove_from_production_succeeds_when_published_once(self):
+        """
+        Tests if can_remove_production() returns true if the item is published only once.
+        """
+
+        with self.app.app_context():
```

Comment:
this shouldn't be required in `test_*` methods as context is set in `SuperdeskTestCase.setUp` method